dask/dataframe/io/parquet/arrow.py

```diff
@@ -1019,12 +1020,17 @@ def write_partition(
                 _meta.set_file_path(filename)
         # Return the schema needed to write the metadata
         if return_metadata:
-            return [{"schema": t.schema, "meta": _meta}]
+            d = {"meta": _meta}
+            if head:
+                # Only return schema if this is the "head" partition
+                d["schema"] = t.schema
+            return [d]
         else:
             return []
 
     @staticmethod
     def write_metadata(parts, fmd, fs, path, append=False, **kwargs):
+        schema = parts[0][0].get("schema", None)
         parts = [p for p in parts if p[0]["meta"] is not None]
```

Comment:
This schema used to be defiled after the parts were filtered for ones that have `"meta"`. Does that change matter?

Reply:
This is a great question - Since we are only instructing the very first partition to return the schema, we can no longer set/write the schema after the empty-metadata partitions are cleared, because it is possible for the very first partition to have empty metadata.

At first I thought this could be a problem, because the schema infered from an empty partition may be "wrong."  However, I eventually realized that, in a situation like this, the schema would need to be specified by the user (or inferable from an empty dataframe) - Otherwise, the following `append_row_groups` operation would fail anyway.

Reply:
Ah ok thanks for explaining. Is there any way to test this change?

Reply:
>Ah ok thanks for explaining. Is there any way to test this change?

I'll think on this a bit more, but itseems  a bit tough to test the change since the end result hasn't changed (and there wasn't really a "bug" fixed).  It's also fairly uncommon for Dask to actually use the `_common_metadata` file.  Perhaps we could be a bit more aggressive about forcing `_common_metadata` to be used in a tricky round-trip situation.

Reply:
Oaky - Still not really sure of a great test here. It doesn't seem like we really "use" the `_common_metadata` file much atm anyway.  However, I added a simple  assertion to test that we are getting the schema right in the case that the first partition is empty.

Reply:
Ok that works!